### PR TITLE
Update design token

### PR DESCRIPTION
**Pull request by Jenesix**
Update design token

### CODE REVIEW

File: src/styles/token.ts [hunk 1]
```diff
@@ -1,7 +1,35 @@
 const DESIGN_TOKEN = {
   color: {
-    primary: '#ff9980',
+    primary: {
```
Comment (Spectrum862) [inline src/styles/token.ts:3]: Select shade for Mui's primary color.

**Review by Spectrum862 — CHANGES_REQUESTED** [review]
run this script and fix.
`pnpm lint`

File: package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "eslint-plugin-unused-imports": "3.1.0",
     "husky": "9.0.11",
     "prettier": "3.2.5",
-    "typescript": "5"
+    "typescript": "^5.4.3"
```
Comment (famesensor) [inline package.json]: please remove ^

File: src/styles/token.ts [hunk 3]
```diff
@@ -1,6 +1,39 @@
...
+      p4: '#290CBA',
+      p5: '#340FED',
+      p6: '#E8E6F3',
+    },
+    secondary: {
+      p1: '#E89582',
+      p2: '#E86F54',
+      p3: '#E94925',
+      p4: '#E8360E',
+      p5: '#4D180C',
```
Comment (famesensor) [inline src/styles/token.ts]: why use p prefix?

**Review by famesensor — CHANGES_REQUESTED** [review]

File: src/styles/token.ts (hunk 3 above)
Comment (Spectrum862) [reply src/styles/token.ts]: Use 1-6 or 100-600 instead.

File: src/styles/theme.ts [hunk 4]
```diff
@@ -40,6 +43,13 @@ const theme = createTheme({
       fontFamily: `${sarabun.style.fontFamily}, "Roboto", "Helvetica", "Arial", sans-serif`,
     },
   },
+  components: {
+    MuiTypography: {
+      defaultProps: {
+        color: DESIGN_TOKEN.color.primary.p1,
```
Comment (Spectrum862) [inline src/styles/theme.ts]: add text primary color in palette and set default color to "text.primary"
```ts
color: 'text.primary'
```

**Review by Spectrum862 — CHANGES_REQUESTED** [review]

File: src/styles/token.ts (hunk 3 above)
Comment (Jenesix) [reply src/styles/token.ts]: when i use 1-6 or 100-600 i don't know why it error when i use in color tag such as
 palette: {
    primary: {
      main: DESIGN_TOKEN.color.primary.3, <----
    },

Comment (Spectrum862) [reply src/styles/token.ts]: fill in [] ->  DESIGN_TOKEN.color.primary[3]

**Review by Spectrum862 — APPROVED** [review]

**Review by famesensor — APPROVED** [review]

**Pull request merged**